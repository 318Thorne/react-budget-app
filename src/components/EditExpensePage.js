import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpense, startRemoveExpense} from '../actions/expenses';


export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
            this.props.onSubmit(this.props.expense.id, expense);
            this.props.history.push('/');
        };
    onClick = () => {
            const id = this.props.expense.id
            this.props.onClick(id);
            this.props.history.push('/');
        };

    render(){
        return (
            <div>
                <ExpenseForm 
                    expense = {this.props.expense}
                    onSubmit={this.onSubmit}                  
                />
                <button onClick={this.onClick}>Remove item</button>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onSubmit: (id, expense) => dispatch(editExpense(id, expense)),
        onClick: (id) => dispatch(startRemoveExpense({id}))       
    }
}

const mapStateToProps = (state, props) => {
    return{
        expense: state.expenses.find((expense) => (expense.id === props.match.params.id) )
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);