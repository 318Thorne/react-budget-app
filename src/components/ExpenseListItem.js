import React from 'react';
import {Link} from 'react-router-dom';


export const ExpenseListItem = ({id , description, amount, createdAt}) => (
    <div>
        <p>{description}</p>
        <p>{amount}</p>
        <p>{createdAt}</p>
        <Link to={`/edit/${id}`}>edit</Link>
        <hr/>
    </div>
);

// const mapStateToProps = (state) => {
//     return {
//         expenses: state.expenses
//     }
// }

export default ExpenseListItem