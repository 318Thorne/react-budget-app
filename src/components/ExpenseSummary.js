import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import ExpenseTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'

export const ExpenseSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses'
    return (
        <div className="page-header">  
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expensesCount}</span> {expenseWord} totalling <span>{numeral(expensesTotal / 100).format('$0,0.00')}</span> </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>          
        </div>
    )
}

const mapStateToProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters)
    return {
        expensesCount: expenses.length,
        expensesTotal: ExpenseTotal(expenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummary);