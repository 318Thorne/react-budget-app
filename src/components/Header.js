import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <ul>
            <li><NavLink to="/" activeClassName="isActive" exact={true}>Home</NavLink></li>
            <li><NavLink to="/create" activeClassName="isActive">Add</NavLink></li>
            <li><NavLink to="/help" activeClassName="isActive" >Help</NavLink></li>
        </ul>
    </header>
);
export default Header;