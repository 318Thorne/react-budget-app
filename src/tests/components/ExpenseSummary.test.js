import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';



test('should render ExpenseSummary component with 1 expense', () => {
    const result = shallow(<ExpenseSummary expensesCount={1} expensesTotal={1000}/>);
    expect(result).toMatchSnapshot();
})

test('should render ExpenseSummary component with expenses', () => {
    const total = expenses[0].amount + expenses[1].amount + expenses[2].amount;
    const result = shallow(<ExpenseSummary expensesCount={expenses} expensesTotal={total} />);
    expect(result).toMatchSnapshot();
})
