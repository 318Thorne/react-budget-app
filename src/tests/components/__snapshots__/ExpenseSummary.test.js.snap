// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary component with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $10.00
      </span>
       
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary component with expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        <Component />
        <Component />
        <Component />
      </span>
       
      expenses
       totalling 
      <span>
        $2,011.96
      </span>
       
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
